fuel/datasets/hdf5.py

```diff
@@ -242,8 +271,16 @@ def load_axis_labels(self):
         handle = self._file_handle
         axis_labels = {}
         for source_name in handle:
-            axis_labels[source_name] = tuple(
-                dim.label for dim in handle[source_name].dims)
+            vlen_source = (len(handle[source_name].dims) == 1
```

Comment:
Also a sanity check could be done, that if `'shapes' in handle.dims[0]`, then `len(handle.dims)` must be one.